### DISCUSSION TITLE
Loading batches in Heterogeneous Graph

### DISCUSSION THREAD

**Discussion by vicbeldo98** [pyg-team/pytorch_geometric#4062] [Q&A] [answered]
Hi! I am a student starting to work with GNN and PyG and I am a bit confused. 
I am trying to build a recommender system using the Yelp Dataset, so my task is link prediction.
The data (which is a Heterogeneous Graph) has the following structure:
![image](https://user-images.githubusercontent.com/49116334/153581801-231d7cc9-ed6f-4a8b-ae80-36efeea7e395.png)

The problem is that it does not fit in memory for training. For that purpose, I have been trying to train with batches. I have tried NeighborLoader and it works only if I specify an input_node_type. The thing is that I feel like I am misunderstanding something, because I want to train with both node_types (user and business) but only one is allowed. Also, I feel like this is not exactly what I wanted, I want randomly removed links, not node removal.

Could someone clarify the situation?

Thank you in advanced,

**Comment by rusty1s** [comment, marked as answer]
Please have a look at https://github.com/pyg-team/pytorch_geometric/issues/4026. We are currently trying to integrate this into PyG :)
